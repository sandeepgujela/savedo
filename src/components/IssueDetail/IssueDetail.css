.issue-detail-container {
  margin-right: auto;
  margin-left: auto;
  width:95%;
  max-width:980px;
}
.issue-detail-container .title{
  font-weight: 400;
}
.issue-detail-container .issue-detail{
  border-radius: 3px ;
  border-top:1px solid #e0e4e8;
  list-style-type: none;
  padding: 0;

}
.status-container {
  margin-bottom:20px;
  display:flex;
  flex-direction: row;
  align-items: center;
}

.status-label-open{
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding:5px;
  border-radius: 3px;
  background-color:#2cbe4e;
  font-weight:600;
  color:white

}


@media only screen and (max-width: 620px) {
  .issue-detail-container .title{
    font-size: 16px;
    line-height: 1.4;
  }
  .status-container {
    font-size: 12px;
  }
}
