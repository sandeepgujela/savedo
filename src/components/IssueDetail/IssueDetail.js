import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { detailsRequest } from '../../state/details/actions';
import Comment from '../Comment/Comment';
import Octicon, { IssueOpened } from '@githubprimer/octicons-react';
import './IssueDetail.css';
import { Redirect } from 'react-router-dom';
import { Issues } from '../Issues/Issues';
import { meaningFullTime } from '../../util/helpers';

class IssueDetail extends Component {
  componentDidMount() {
    const { selected, dispatch } = this.props;
    if (selected === undefined) {
      return;
    }
    dispatch(detailsRequest({ url: selected.comments_url }));
  }

  static getTitle(selected) {
    const { title, number } = selected;
    return (
      <div>
        <span>
          <h1 className="title">
            {title} <span className="color-light">#{number}</span>
          </h1>
        </span>
      </div>
    );
  }

  static getSubtitle(selected) {
    const {
      user: { login },
      created_at,
      comments,
    } = selected;
    return (
      <div className="status-container">
        <div className="status-label-open">
          <Octicon icon={IssueOpened} size="small" verticalAlign="middle" />{' '}
          &nbsp;Open
        </div>
        <div className="ml-5 color-light">
          <b>{login}</b> opened this issue on {meaningFullTime(created_at)} .{' '}
          {comments} Comments
        </div>
      </div>
    );
  }

  render() {
    const { selected, data } = this.props;
    if (selected === undefined || !selected.number) {
      return <Redirect to={'/'} />;
    }
    return (
      <div className="issue-detail-container">
        {Issues.getPageTitle()}
        {IssueDetail.getTitle(selected)}
        {IssueDetail.getSubtitle(selected)}
        <Comment data={selected} />
        {data &&
          data.length > 0 &&
          data.map(comment => <Comment key={comment.id} data={comment} />)}
      </div>
    );
  }
}

IssueDetail.propTypes = {
  selected: PropTypes.shape({
    title: PropTypes.string.isRequired,
    number: PropTypes.number.isRequired,
    created_at: PropTypes.string.isRequired,
    html_url: PropTypes.string.isRequired,
    state: PropTypes.string.isRequired,
    labels: PropTypes.arrayOf(
      PropTypes.shape({
        color: PropTypes.string,
        name: PropTypes.string,
      }),
    ).isRequired,
    comments: PropTypes.number.isRequired,
    user: PropTypes.shape({
      login: PropTypes.string.isRequired,
      avatar_url: PropTypes.string,
    }),
  }).isRequired,
};

const mapsStateToProps = state => {
  const { selected } = state.issues;
  const { data } = state.details;
  return {
    selected,
    data,
  };
};

const IssueDetailConnected = connect(mapsStateToProps)(IssueDetail);

export { IssueDetail, IssueDetailConnected };
