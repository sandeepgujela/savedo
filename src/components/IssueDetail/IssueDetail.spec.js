import React from 'react';
import { shallow } from 'enzyme';
import { IssueDetail } from './IssueDetail';

describe('<IssueDetail />', () => {
  const build = passedProps => {
    const defaultProps = {
      match: {
        params: {
          issueId: 123,
        },
      },
      selected: {
        html_url: 'some_url',
        state: 'open',
        labels: [],
        number: 123,
        title: 'Some Title',
        comments: 1,
        user: {
          login: 'savedo',
          avatar_url: 'some-url',
        },
        created_at: 'xx.yy.zzzz',
        body: 'comment',
      },
      data: [
        {
          id: 1234,
          user: {
            login: 'savedo-commentor',
            avatar_url: 'some-url',
          },
          created_at: 'xx.yy.zzzz',
          body: 'some comment',
        },
      ],
    };
    return shallow(<IssueDetail {...defaultProps} {...passedProps} />);
  };

  it('should render', () => {
    const dispatch = jest.fn();
    const component = build({ dispatch });
    expect(component).toMatchSnapshot();
  });
});
