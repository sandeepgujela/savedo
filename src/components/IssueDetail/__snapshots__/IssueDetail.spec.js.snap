// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IssueDetail /> should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IssueDetail
    data={
      Array [
        Object {
          "body": "some comment",
          "created_at": "xx.yy.zzzz",
          "id": 1234,
          "user": Object {
            "avatar_url": "some-url",
            "login": "savedo-commentor",
          },
        },
      ]
    }
    dispatch={
      [MockFunction] {
        "calls": Array [
          Array [
            [Function],
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    match={
      Object {
        "params": Object {
          "issueId": 123,
        },
      }
    }
    selected={
      Object {
        "body": "comment",
        "comments": 1,
        "created_at": "xx.yy.zzzz",
        "html_url": "some_url",
        "labels": Array [],
        "number": 123,
        "state": "open",
        "title": "Some Title",
        "user": Object {
          "avatar_url": "some-url",
          "login": "savedo",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="home"
          replace={false}
          to="/"
        >
          <span>
            <Octicon
              className="octicon"
              icon={[Function]}
              size="small"
              verticalAlign="middle"
            />
          </span>
          <span
            className=" ml-5 project-title color-primary"
          >
            facebook
            <b>
              react
            </b>
          </span>
        </Link>,
        <div>
          <span>
            <h1
              className="title"
            >
              Some Title
               
              <span
                className="color-light"
              >
                #
                123
              </span>
            </h1>
          </span>
        </div>,
        <div
          className="status-container"
        >
          <div
            className="status-label-open"
          >
            <Octicon
              className="octicon"
              icon={[Function]}
              size="small"
              verticalAlign="middle"
            />
             
             Open
          </div>
          <div
            className="ml-5 color-light"
          >
            <b>
              savedo
            </b>
             opened this issue on 
            just now
             .
             
            1
             Comments
          </div>
        </div>,
        <Comment
          data={
            Object {
              "body": "comment",
              "comments": 1,
              "created_at": "xx.yy.zzzz",
              "html_url": "some_url",
              "labels": Array [],
              "number": 123,
              "state": "open",
              "title": "Some Title",
              "user": Object {
                "avatar_url": "some-url",
                "login": "savedo",
              },
            }
          }
        />,
        Array [
          <Comment
            data={
              Object {
                "body": "some comment",
                "created_at": "xx.yy.zzzz",
                "id": 1234,
                "user": Object {
                  "avatar_url": "some-url",
                  "login": "savedo-commentor",
                },
              }
            }
          />,
        ],
      ],
      "className": "issue-detail-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <span>
              <Octicon
                className="octicon"
                icon={[Function]}
                size="small"
                verticalAlign="middle"
              />
            </span>,
            <span
              className=" ml-5 project-title color-primary"
            >
              facebook
              <b>
                react
              </b>
            </span>,
          ],
          "className": "home",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Octicon
                className="octicon"
                icon={[Function]}
                size="small"
                verticalAlign="middle"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "octicon",
                "icon": [Function],
                "size": "small",
                "verticalAlign": "middle",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "facebook",
                <b>
                  react
                </b>,
              ],
              "className": " ml-5 project-title color-primary",
            },
            "ref": null,
            "rendered": Array [
              "facebook",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "react",
                },
                "ref": null,
                "rendered": "react",
                "type": "b",
              },
            ],
            "type": "span",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            <h1
              className="title"
            >
              Some Title
               
              <span
                className="color-light"
              >
                #
                123
              </span>
            </h1>
          </span>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="title"
            >
              Some Title
               
              <span
                className="color-light"
              >
                #
                123
              </span>
            </h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Some Title",
                " ",
                <span
                  className="color-light"
                >
                  #
                  123
                </span>,
              ],
              "className": "title",
            },
            "ref": null,
            "rendered": Array [
              "Some Title",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "#",
                    123,
                  ],
                  "className": "color-light",
                },
                "ref": null,
                "rendered": Array [
                  "#",
                  123,
                ],
                "type": "span",
              },
            ],
            "type": "h1",
          },
          "type": "span",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="status-label-open"
            >
              <Octicon
                className="octicon"
                icon={[Function]}
                size="small"
                verticalAlign="middle"
              />
               
               Open
            </div>,
            <div
              className="ml-5 color-light"
            >
              <b>
                savedo
              </b>
               opened this issue on 
              just now
               .
               
              1
               Comments
            </div>,
          ],
          "className": "status-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Octicon
                  className="octicon"
                  icon={[Function]}
                  size="small"
                  verticalAlign="middle"
                />,
                " ",
                " Open",
              ],
              "className": "status-label-open",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "octicon",
                  "icon": [Function],
                  "size": "small",
                  "verticalAlign": "middle",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              " ",
              " Open",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  savedo
                </b>,
                " opened this issue on ",
                "just now",
                " .",
                " ",
                1,
                " Comments",
              ],
              "className": "ml-5 color-light",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "savedo",
                },
                "ref": null,
                "rendered": "savedo",
                "type": "b",
              },
              " opened this issue on ",
              "just now",
              " .",
              " ",
              1,
              " Comments",
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Object {
            "body": "comment",
            "comments": 1,
            "created_at": "xx.yy.zzzz",
            "html_url": "some_url",
            "labels": Array [],
            "number": 123,
            "state": "open",
            "title": "Some Title",
            "user": Object {
              "avatar_url": "some-url",
              "login": "savedo",
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1234",
        "nodeType": "class",
        "props": Object {
          "data": Object {
            "body": "some comment",
            "created_at": "xx.yy.zzzz",
            "id": 1234,
            "user": Object {
              "avatar_url": "some-url",
              "login": "savedo-commentor",
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="home"
            replace={false}
            to="/"
          >
            <span>
              <Octicon
                className="octicon"
                icon={[Function]}
                size="small"
                verticalAlign="middle"
              />
            </span>
            <span
              className=" ml-5 project-title color-primary"
            >
              facebook
              <b>
                react
              </b>
            </span>
          </Link>,
          <div>
            <span>
              <h1
                className="title"
              >
                Some Title
                 
                <span
                  className="color-light"
                >
                  #
                  123
                </span>
              </h1>
            </span>
          </div>,
          <div
            className="status-container"
          >
            <div
              className="status-label-open"
            >
              <Octicon
                className="octicon"
                icon={[Function]}
                size="small"
                verticalAlign="middle"
              />
               
               Open
            </div>
            <div
              className="ml-5 color-light"
            >
              <b>
                savedo
              </b>
               opened this issue on 
              just now
               .
               
              1
               Comments
            </div>
          </div>,
          <Comment
            data={
              Object {
                "body": "comment",
                "comments": 1,
                "created_at": "xx.yy.zzzz",
                "html_url": "some_url",
                "labels": Array [],
                "number": 123,
                "state": "open",
                "title": "Some Title",
                "user": Object {
                  "avatar_url": "some-url",
                  "login": "savedo",
                },
              }
            }
          />,
          Array [
            <Comment
              data={
                Object {
                  "body": "some comment",
                  "created_at": "xx.yy.zzzz",
                  "id": 1234,
                  "user": Object {
                    "avatar_url": "some-url",
                    "login": "savedo-commentor",
                  },
                }
              }
            />,
          ],
        ],
        "className": "issue-detail-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <span>
                <Octicon
                  className="octicon"
                  icon={[Function]}
                  size="small"
                  verticalAlign="middle"
                />
              </span>,
              <span
                className=" ml-5 project-title color-primary"
              >
                facebook
                <b>
                  react
                </b>
              </span>,
            ],
            "className": "home",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Octicon
                  className="octicon"
                  icon={[Function]}
                  size="small"
                  verticalAlign="middle"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "octicon",
                  "icon": [Function],
                  "size": "small",
                  "verticalAlign": "middle",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "facebook",
                  <b>
                    react
                  </b>,
                ],
                "className": " ml-5 project-title color-primary",
              },
              "ref": null,
              "rendered": Array [
                "facebook",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "react",
                  },
                  "ref": null,
                  "rendered": "react",
                  "type": "b",
                },
              ],
              "type": "span",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              <h1
                className="title"
              >
                Some Title
                 
                <span
                  className="color-light"
                >
                  #
                  123
                </span>
              </h1>
            </span>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1
                className="title"
              >
                Some Title
                 
                <span
                  className="color-light"
                >
                  #
                  123
                </span>
              </h1>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Some Title",
                  " ",
                  <span
                    className="color-light"
                  >
                    #
                    123
                  </span>,
                ],
                "className": "title",
              },
              "ref": null,
              "rendered": Array [
                "Some Title",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "#",
                      123,
                    ],
                    "className": "color-light",
                  },
                  "ref": null,
                  "rendered": Array [
                    "#",
                    123,
                  ],
                  "type": "span",
                },
              ],
              "type": "h1",
            },
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="status-label-open"
              >
                <Octicon
                  className="octicon"
                  icon={[Function]}
                  size="small"
                  verticalAlign="middle"
                />
                 
                 Open
              </div>,
              <div
                className="ml-5 color-light"
              >
                <b>
                  savedo
                </b>
                 opened this issue on 
                just now
                 .
                 
                1
                 Comments
              </div>,
            ],
            "className": "status-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Octicon
                    className="octicon"
                    icon={[Function]}
                    size="small"
                    verticalAlign="middle"
                  />,
                  " ",
                  " Open",
                ],
                "className": "status-label-open",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "octicon",
                    "icon": [Function],
                    "size": "small",
                    "verticalAlign": "middle",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                " ",
                " Open",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    savedo
                  </b>,
                  " opened this issue on ",
                  "just now",
                  " .",
                  " ",
                  1,
                  " Comments",
                ],
                "className": "ml-5 color-light",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "savedo",
                  },
                  "ref": null,
                  "rendered": "savedo",
                  "type": "b",
                },
                " opened this issue on ",
                "just now",
                " .",
                " ",
                1,
                " Comments",
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Object {
              "body": "comment",
              "comments": 1,
              "created_at": "xx.yy.zzzz",
              "html_url": "some_url",
              "labels": Array [],
              "number": 123,
              "state": "open",
              "title": "Some Title",
              "user": Object {
                "avatar_url": "some-url",
                "login": "savedo",
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1234",
          "nodeType": "class",
          "props": Object {
            "data": Object {
              "body": "some comment",
              "created_at": "xx.yy.zzzz",
              "id": 1234,
              "user": Object {
                "avatar_url": "some-url",
                "login": "savedo-commentor",
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
